### ISSUE TITLE
compiletest should note when test output is normalized

### ISSUE THREAD

**Issue by jyn514** [rust-lang/rust#125537]
compiletest has two basic kinds of normalization:
1. normalization it does itself, either built-in or through a `normalize-stderr` pattern. this is what the UI suite uses.
2. normalization it doesn't know about, e.g. in a run-make test

normalizing outputs can lead to some extremely confusing errors, such as https://github.com/rust-lang/rust/pull/121571#issuecomment-2129928216. to avoid misleading contributors, it should note when the output has been normalized.

the first part is not too hard; it can save list of lines that have been modified in each test and print something like "note: 32/32 lines that did not match were normalized before comparison; see build/test/ui/foo.out.orig for unnormalized version"
(note that the unnormalized file already exists today but almost nobody knows about it).

the second part is trickier; run-make tests often run sed before comparing, and this is opaque to compiletest. perhaps fixing this can be part of the ongoing work to switch run-make to rust? and, in the meantime, introduce a new SED script, based off cut-and-grep.sh, to prevent the problem from getting worse? cc @jieyouxu

rustbot label A-contributor-roadblock A-diagnostics D-confusing A-testsuite

**Comment by jieyouxu** [comment]
> run-make tests often run sed before comparing, and this is opaque to compiletest. perhaps fixing this can be part of the ongoing work to switch run-make to rust? and, in the meantime, introduce a new SED script, based off cut-and-grep.sh, to prevent the problem from getting worse?

yes, if we're porting any run-make test that relies on some kind of normalization on some output, we should definitely make the error output as helpful as possible (i.e. a helper in the support lib), that basically on top of reporting what post-normalization check failed, also report what the pre-normalization output was.

**Comment by jyn514** [comment]
https://github.com/rust-lang/rust/pull/133733 does the UI test side of this. the run-make bit should be simpler, actually, since the codebase is much less complicated. it just needs a couple lines around https://github.com/rust-lang/rust/blob/50fd83cfa64ce82ed1cbb118ea168337b907c92f/src/tools/run-make-support/src/diff/mod.rs#L96-L118 i think
